From 9a9a74da1641243b9c7420a9e27653ed3d0b953a Mon Sep 17 00:00:00 2001
From: Tao Huang <huangtao@rock-chips.com>
Date: Thu, 9 Apr 2020 15:32:45 +0800
Subject: [PATCH] rk: kbuild: Introduce CONFIG_DTC_SYMBOLS

dtc generation of symbols by CONFIG_DTC_SYMBOLS other than
CONFIG_ANDROID directly.

Fixes: 761bc9429983 ("ANDROID: rk: kbuild: dtc generation of symbols")
Change-Id: Ia10496031bc02fd3a4ff98ab0acfc6fc9a54951b
Signed-off-by: Tao Huang <huangtao@rock-chips.com>
---
 drivers/of/Kconfig   | 8 ++++++++
 scripts/Makefile.lib | 2 +-
 2 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/drivers/of/Kconfig b/drivers/of/Kconfig
index 6953974e9d96..485d4f736fbf 100644
--- a/drivers/of/Kconfig
+++ b/drivers/of/Kconfig
@@ -12,6 +12,14 @@ menuconfig OF
 
 if OF
 
+config DTC_SYMBOLS
+	bool "Enable dtc generation of symbols for overlays support"
+	depends on DTC
+	default ANDROID
+	help
+	  Set DTC_FLAGS += -@
+	  Android OS must enable this option.
+
 config DTC_OMIT_DISABLED
 	bool "Fixup dtb by removing nodes with disabled status"
 	depends on DTC
diff --git a/scripts/Makefile.lib b/scripts/Makefile.lib
index 47a24a6863ad..3639de2d3142 100644
--- a/scripts/Makefile.lib
+++ b/scripts/Makefile.lib
@@ -245,7 +245,7 @@ cmd_gzip = (cat $(filter-out FORCE,$^) | gzip -n -f -9 > $@) || \
 DTC ?= $(objtree)/scripts/dtc/dtc
 
 # Generation of symbols for Android
-ifeq ($(CONFIG_ANDROID),y)
+ifeq ($(CONFIG_DTC_SYMBOLS),y)
 DTC_FLAGS += -@
 endif
 
-- 
2.35.3

